test/metadata-stream.test.js

```diff
@@ -518,6 +518,27 @@ QUnit.test('constructs the dispatch type', function(assert) {
   assert.equal(metadataStream.dispatchType, '1503020100', 'built the dispatch type');
 });
 
+QUnit.test('should skip tag frame parsing on malformed frame, preserving previous frames', function(assert) {
+  var events = [],
+      validFrame = id3Frame('TIT2',
+                            0x03, // utf-8
+                            stringToCString('sample title')),
+      malformedFrame = id3Frame('WOAF'), // frame with size of 0B
```

Comment:
I could also create the same malformed frame like this:
```
stringToInts('WOAF').concat([
        0x00, 0x00, 0x00, 0x00, // size
        0x00, 0x00 // flags
      ]),
```
to make the malformation more visible